### PR TITLE
Add news demo

### PR DESCRIPTION
**Pull request by goncy**
Adds [this demo](https://next-news.vercel.app) to the list

### CODE REVIEW

File: edge-functions/next-news/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Hacker Next
+
+This is an example showing a hacker news clone using Next.js with edge functions
+
+## Demo
+
+https://edge-next-news.vercel.app
```
Comment (lfades) [inline edge-functions/next-news/README.md:7]: This is the same prefix other examples are using

```suggestion
https://edge-functions-next-news.vercel.app
```

File: edge-functions/next-news/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+### Clone and Deploy
+
+Execute [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app) with [npm](https://docs.npmjs.com/cli/init) or [Yarn](https://yarnpkg.com/lang/en/docs/cli/create/) to bootstrap the example:
+
+```bash
+npx create-next-app --example https://github.com/vercel/examples/tree/main/edge-functions/edge-next-news edge-next-news
+# or
+yarn create next-app --example https://github.com/vercel/examples/tree/main/edge-functions/edge-next-news edge-next-news
+```
```
Comment (lfades) [inline edge-functions/next-news/README.md:27]: The naming here should change to `next-news` from `edge-next-news`

File: edge-functions/next-news/components/header.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import Nav from "./nav";
+import Logo from "./logo";
+import Link from "next/link";
+
+export default () => (
```
Comment (lfades) [inline edge-functions/next-news/components/header.tsx:5]: Let's export named components, Next.js should be complaining about this in the console.

File: edge-functions/next-news/pages/_middleware.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { NextResponse as Response } from "next/server";
+
+export default function middleware(req) {
+  if (req.nextUrl.pathname === "/") {
+    return Response.rewrite("/news/1");
+  }
+}
```
Comment (lfades) [inline edge-functions/next-news/pages/_middleware.ts:7]: This part should be included on the readme, alongside a description of why are we using middleware in this example.

**Review by lfades — CHANGES_REQUESTED** [review]
Added some comments

File: edge-functions/next-news/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# Hacker Next
```
Comment (leerob) [inline edge-functions/next-news/README.md:1]:
```suggestion
# Next News
```

File: edge-functions/next-news/.gitignore [hunk 6]
```diff
@@ -0,0 +1,5 @@
+node_modules
+.next
+
+.vercel
+.output
```
Comment (lfades) [inline edge-functions/next-news/.gitignore:5]: Let's use the .gitignore of our other examples here.

File: edge-functions/next-news/README.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+export default function middleware(req) {
+  if (req.nextUrl.pathname === "/") {
+    return Response.rewrite("/news/1");
+  }
+}
+```
+> This way you can rewrite the home page to the first news
+
+This is an example showing a hacker news clone using Next.js with edge functions
```
Comment (lfades) [inline edge-functions/next-news/README.md:16]:
````suggestion

This is an example showing a hacker news clone using Next.js. Edge functions are used to show how a rewrite from `/` to `/news/1` works. Below is the code from [pages/\_middleware.ts](pages/_middleware.ts):

```ts
import { NextResponse as Response } from "next/server";

export default function middleware(req) {
  if (req.nextUrl.pathname === "/") {
    return Response.rewrite("/news/1");
  }
}
```
````

File: edge-functions/next-news/README.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
+# Next News
+
+
```
Comment (lfades) [inline edge-functions/next-news/README.md:3]:
```suggestion
```

**Review by lfades — APPROVED** [review]
LGTM!

**Pull request merged**